src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildANetCoreApp.cs

```diff
@@ -562,5 +602,54 @@ public void It_uses_lowercase_form_of_the_target_framework_for_the_output_path()
                 .Should()
                 .BeEquivalentTo("netcoreapp1.1");
         }
+
+        private static unsafe bool IsPE32(string path)
+        {
+            const UInt16 PEFileSignature = 0x5A4D;
+            const int PEHeaderPointerOffset = 0x3C;
+            const int MagicOffset = 0x18;
+            const int PE32Magic = 0x10B;
+            const int PE32PlusMagic = 0x20B;
+
+            using (var memoryMappedFile = MemoryMappedFile.CreateFromFile(path))
```

Comment:
Consider using System.Reflection.PortableExecutable namespace in System.Reflection.Medata.dll in the BCL for this:

``` c#
using (var peReader = new PEReader(File.OpenRead(path)))
{
    switch (peReader.PEHeaders.PEHeader.Magic)
    {
        ...
    }
}
```

Reply:
Completely ignorant that existed.  We should probably use it for the apphost customization (i.e. changing the subsystem in the PE header) too.

I'll see what I can do to use the `PEReader`.

Reply:
Aside: There's a PEBuilder too for writing. Roslyn emits with it, but I don't know that it would easy (or even possible) to do the apphost transformation with it. We do have an open issue to avoid the Win32 API so there might be some things helpful in there. In a few *simple* cases, we've managed to use just the reader to read some data then seek to an offset and patch a few bits. For example, before the compiler had public signing we had a little tool to do it after the fact this way.

Reply:
After looking at the docs, the builder seems too heavy for that use case, and there's no easy way to transform a reader to a builder to make simple modifications from what I can tell.  I'd agree that it would be overkill to just toggle a byte at a well-known offset in the file.